kernel/lib/runlevel/5_operation/influxdb2.go

```diff
@@ -0,0 +1,224 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package operation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/influxdata/influxdb-client-go/v2/api/write"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/fablab/kernel/model"
+	"github.com/openziti/foundation/v2/errorz"
+	log "github.com/sirupsen/logrus"
+	"net/url"
+	"time"
+)
+
+func InfluxMetricsReporter2() model.Stage {
+	return &influxMetricsReporterStage2{}
+}
+
+type influxMetricsReporterStage2 struct {
+	errorz.ErrorHolderImpl
+}
+
+func (stage *influxMetricsReporterStage2) Execute(run model.Run) error {
+	m := run.GetModel()
+	urlVar := m.GetRequiredStringVariable(stage, "metrics.influxdb.url")
+	db := m.GetRequiredStringVariable(stage, "metrics.influxdb.db")
+	token := m.GetRequiredStringVariable(stage, "metrics.influxdb.token")
+	org := m.GetRequiredStringVariable(stage, "metrics.influxdb.org")
+	bucket := m.GetRequiredStringVariable(stage, "metrics.influxdb.bucket")
+
+	if stage.HasError() {
+		return stage.GetError()
+	}
+
+	influxUrl, err := url.Parse(urlVar)
+	if err != nil {
+		return err
+	}
+
+	config := influxConfig2{
+		url:      *influxUrl,
+		database: db,
+		token:    token,
+		org:      org,
+		bucket:   bucket,
+	}
+
+	handler, err := NewInfluxDBMetricsHandler2(&config)
+	if err != nil {
+		return err
+	}
+
+	m.MetricsHandlers = append(m.MetricsHandlers, handler)
+	return nil
+}
+
+func NewInfluxDBMetricsHandler2(cfg *influxConfig2) (model.MetricsHandler, error) {
+	rep := &influxReporter2{
+		url:         cfg.url,
+		database:    cfg.bucket,
+		token:       cfg.token,
+		metricsChan: make(chan *hostMetricsEvent2, 10),
+		org:         cfg.org,
+	}
+
+	if err := rep.makeClient(); err != nil {
+		return nil, fmt.Errorf("unable to make HandlerTypeInfluxDB influxdb. err=%v", err)
+	}
+
+	go rep.run()
+	return rep, nil
+}
+
+type influxReporter2 struct {
+	url         url.URL
+	database    string
+	metricsChan chan *hostMetricsEvent2
+	client      influxdb2.Client
+	token       string
+	org         string
+	bucket      string
```

Comment:
linter is reporting that bucket is unused. Remove if not needed

Reply:
Removed.